index.js

```diff
@@ -12,8 +12,8 @@ const DEFAULT_403_PAGE = path.join(__dirname, 'static', '403.html')
 const DEFAULT_404_PAGE = path.join(__dirname, 'static', '404.html')
 
 function fastifyStatic (fastify, opts, next) {
-  const error = checkOptions(opts)
-  if (error instanceof Error) return next(error)
+  const error = checkRootPathForErrors(opts.root)
```

Comment:
I'd like to keep this function generic in order to validate all parameters

We should check the `opts.page*Path` too if present